Modules/.AL-Go/CompileAppInBcContainer.ps1

```diff
@@ -2,40 +2,42 @@ Param(
     [Hashtable] $parameters
 )
 
+# Setup compiler features to generate captions and LCGs
 if (!$parameters.ContainsKey("Features")) {
     $parameters["Features"] = @()
 }
-$parameters["Features"] = @("lcgtranslationfile", "generateCaptions")
+$parameters["Features"] += @("lcgtranslationfile", "generateCaptions")
 
 $appFile = Compile-AppInBcContainer @parameters
 
-if ($appFile) {
-    $filename = [System.IO.Path]::GetFileName($appFile)
-    if ($filename -like "Microsoft_System Application_*.*.*.*.app") {
-        # System application compiled - add BaseApp and Application app from container to output
-        Invoke-ScriptInBcContainer -containerName $parameters.ContainerName -scriptblock { Param([string]$packagesFolder)
-            function CopyApp {
-                Param(
-                    [string] $localAppPath,
-                    [string] $w1AppPath
-                )
-
-                $name = [System.IO.Path]::GetFileName($w1AppPath)
-                Write-Host "Copying $name to packages path"
-                if (Test-Path $localAppPath) {
-                    $appPath = $localAppPath
-                }
-                else {
-                    $appPath = $w1AppPath
-                }
-                Copy-Item -Path $appPath -Destination (Join-Path $packagesFolder $name)
-                Copy-Item -Path $appPath -Destination (Join-Path "c:\run\my" $name)
-            }
-
-            CopyApp -localAppPath "C:\Applications.*\Microsoft_Base Application_*.*.*.*.app"    -w1AppPath "C:\Applications\BaseApp\Source\Microsoft_Base Application.app"
-            CopyApp -localAppPath "C:\Applications.*\Microsoft_Application_*.*.*.*.app"         -w1AppPath "C:\Applications\Application\Source\Microsoft_Application.app" 
-        } -argumentList (Get-BcContainerPath -ContainerName $parameters.ContainerName -path $Parameters.appSymbolsFolder)
+$branchName = $ENV:GITHUB_REF_NAME
+
```

Comment:
You should include this code here:

$workflowName = "$env:GITHUB_WORKFLOW".Trim()
And then check that workflowName -eq "CI/CD"

Else the code will also trigger on other workflows.

Reply:
Maybe also check $ENV:GITHUB_REPOSITORY is microsoft/ALAppExtensions (to ensure the code isn't run on forks)

Reply:
I'd imagine if no deliver is needed in the fork, there won't be a delivery context.

Reply:
You are right on that - unless you have the secret in a shared KeyVault - but probably a non-problem